lint/nestingReduce_checker.go

```diff
@@ -0,0 +1,70 @@
+package lint
+
+import "go/ast"
+
+//! Finds where nesting level could be reduced.
+//
+// @Before:
+// for _, v := range a {
+//    if v.Bool {
+//        ...
+//    }
+// }
+//
+// @After:
+// for _, v := range a {
+//    if ! v.Bool {
+//        continue
+//    }
+//    ...
+// }
+
+func init() {
+	addChecker(&nestingReduceChecker{}, attrExperimental)
+}
+
+type nestingReduceChecker struct {
+	checkerBase
+}
+
+func (c *nestingReduceChecker) VisitFuncDecl(decl *ast.FuncDecl) {
+	if decl.Body == nil {
+		return
+	}
```

Comment:
Doesn't checkerBase provide nil body filters by default?
Now when I think of it, maybe we should check other `FuncDeclVisitor` implementations and see whether they handle external functions or not. They usually should if they are only interested in function signature and not function body.

Reply:
tl;dr: this nil check is redundant here, but we may miss `EnterFunc` override in some other checkers.